### ISSUE TITLE
Migrating configuration to nginx?

### ISSUE THREAD

**Issue by treii28** [certbot/certbot#5675]
Not so much a bug but just a learning curve issue. I inherited a server set up on apache2 and recently had to migrate it to a varnish cache with the actual https content being served by nginx.  The copy of magento was also upgraded a couple of times and I believe, in the process, some of the files and settings got lost. (/srv/root/.well-known/acme-challenge and the htaccess location settings for it for example)

## My operating system is (include version):
Ubuntu 16.04

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
I believe this was installed with certbot-auto (on the original apache https configuration)

## I ran this command and it produced this output:
I'm trying to migrate it over to using nginx. Running:

> certbot renew --pre-hook "service nginx stop" --post-hook "service nginx start"

after updating the /etc/letsencrypt/renewal/ourserver.conf to use nginx as authenticator results in a timeout error failing to renew the certificate

## Certbot's behavior differed from what I expected because:
I'm new to this so I don't know what to expect. If those files were lost, then it's behaving as I would expect. I just need to know how to fix it. Apparently my predecessor already set up the apache server so there are already certificates, but most of the instructions I'm finding seem to start from not having one. So I'm unclear what needs to be done to just change the renewal (and any other settings) to using nginx and not apache and what files may need to be created or what entries need to be added to the nginx config to allow access to them

Basically I need to know either what command to run to create and/or what files/settings to add into the nginx server root and config files to allow the renew process to work in it rather than in apache. As mentioned the original server root contents were replaced and I'm trying to look for my back-up copies but as this is a new install, and I've had to change the versions/configurations a number of times, I'm having to scour through multiple copies to see if I can find the original version somewhere with the original files.

**Comment by treii28** [comment]
Update:

I found some instructions that may or may not be correct regarding the 'well-known' file so I created a directory in the server root with the permissions of the nginx daemon as .../.well-known/acme-challenge/ and placed a file called 'test' in it with the contents 'success'

I then added a location in the https server configuration block for nginx as:

```
   # letsencrypt challenge
    location ~ /.well-known {
        allow all;
    }

```
I restarted and re-ran the certbot renew command in the original post, but still get the same failure:

```
> certbot renew --pre-hook "service nginx stop" --post-hook "service nginx start"
Saving debug log to /var/log/letsencrypt/letsencrypt.log

-------------------------------------------------------------------------------
Processing /etc/letsencrypt/renewal/www.albumenvy.com.conf
-------------------------------------------------------------------------------
Cert is due for renewal, auto-renewing...
Plugins selected: Authenticator nginx, Installer nginx
Running pre-hook command: service nginx stop
Renewing an existing certificate
Performing the following challenges:
tls-sni-01 challenge for www.albumenvy.com
tls-sni-01 challenge for albumenvy.com
nginx: [error] open() "/run/nginx.pid" failed (2: No such file or directory)
Waiting for verification...
Cleaning up challenges
Attempting to renew cert (www.albumenvy.com) from /etc/letsencrypt/renewal/www.$
lbumenvy.com.conf produced an unexpected error: Failed authorization procedure. 
www.albumenvy.com (tls-sni-01): urn:acme:error:connection :: The server could n$t connect to the client to verify the domain :: Timeout. Skipping.
All renewal attempts failed. The following certs could not be renewed:
  /etc/letsencrypt/live/www.albumenvy.com/fullchain.pem (failure)

-------------------------------------------------------------------------------

All renewal attempts failed. The following certs could not be renewed:
  /etc/letsencrypt/live/www.albumenvy.com/fullchain.pem (failure)
-------------------------------------------------------------------------------
Running post-hook command: service nginx start
Hook command "service nginx start" returned error code 1
Error output from service:
Job for nginx.service failed because the control process exited with error code$ See "systemctl status nginx.service" and "journalctl -xe" for details.

1 renew failure(s), 0 parse failure(s)

IMPORTANT NOTES:
 - The following errors were reported by the server:

   Domain: www.albumenvy.com
   Type:   connection
   Detail: Timeout

   To fix these errors, please make sure that your domain name was
   entered correctly and the DNS A/AAAA record(s) for that domain
   contain(s) the right IP address. Additionally, please check that
   your computer has a publicly routable IP address and that no
   firewalls are preventing the server from communicating with the
   client. If you're using the webroot plugin, you should also verify
   that you are serving files from the webroot path you provided.

```

**Comment by ohemorange** [comment]
Hi @treii28. Since it doesn't sound like this is a bug, I'm going to recommend asking this over at our community forums, https://community.letsencrypt.org/, where a much larger community of people familiar with the project will be able to answer your questions quickly and in greater depth. If this is in fact a bug, let me know and I'll reopen.

**Comment by treii28** [comment]
ah ok, I didn't know there was a forum. Thanks. I think I isolated the problem though - granted, I still needed to do what I did above but a very restrictive firewall was preventing the auto-renew from working properly.  Problems with the lack of ssl on apache was what was preventing it from restarting. I'll see now if the problems with the firewall preventing the renewal check stop nginx tonight. But otherwise, I think my migration is done. You might want to include better configuration details in the instructions that don't rely on someone not yet having a certificate.
